### DISCUSSION TITLE
Remove debugging symbols from production app

### DISCUSSION THREAD

**Discussion by trevor-e** [ghostty-org/ghostty#3538] [Feature Requests, Ideas] [answered]
Hey folks,

I noticed the app currently ships with some debugging symbols included. For example, if you run:
```
strip -rSTx /Applications/Ghostty.app/Contents/MacOS/ghostty -o ~/Documents/ghostty-stripped

➜  ~ git:(master) ✗ ls -l /Applications/Ghostty.app/Contents/MacOS/ghostty
-rwxr-xr-x@ 1 telkins  admin  41046960 Dec 26 15:07 /Applications/Ghostty.app/Contents/MacOS/ghostty

➜  ~ git:(master) ✗ ls -l ~/Documents/ghostty-stripped
-rwxr-xr-x@ 1 telkins  staff  35594912 Dec 27 13:26 /Users/telkins/Documents/ghostty-stripped
```
you can see the stripped version of the binary is 5452048 bytes smaller (~5.5MB). Pretty quick win to make the binary 13% smaller.  Are these symbols needed by anything? I saw there's a built-in crash reporter, not totally sure how that works, but you can typically symbolicate crash reports using separate files for debugging symbols, e.g. `.dSYM` files on macOS.

Note: you also should strip any libraries included in the `Frameworks/` folder, although right now that's only `Sparkle.framework`.

**Comment by mitchellh** [comment]
We should strip them, we save official dSYMs so we can remap symbols.

**Comment by mitchellh** [comment, marked as answer]
We've had some value with people getting symbolized traces using `sample` and so on, so I'm going to keep them for now.
